### PR TITLE
Fixes PS 1.7 PASSWORD_LENGTH #10475

### LINKED ISSUE #10475: PS 1.7 PASSWORD_LENGTH
**Describe the bug**
Hi, I've tried to change PASSWORD_LENGTH into classes/Validate.php (to 8 for example). But when registering a new customer with Classic theme, the allowed length is still 5. I had a look into the html code, on the register page, and saw a pattern=".{5,}". If I change the value 5 by 1, I'm able to register a new customer with a password of 1 character. So it seems that function isPasswd(...) is never used on the controller, because there is no check on the backend of the length of the password.

**To Reproduce**
1. Try changing password_length to 8 in classes/Validate.php
2. Try registering a new customer with a 5 char password
3. it will work

**Screenshots**
If applicable, add screenshots or screenrecords to help explain your problem.

**Additionnal information**
PrestaShop version: 1.7.4.2
PHP version: 7.0.31

### PR DESCRIPTION
**Pull request by web-fu**
Added password validation

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Solve problem of mismatch in password validation
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | No
| Deprecations? |No
| Fixed ticket? | Fixes #10475
| How to test?  | Change password length

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12340)

### CODE REVIEW

File: classes/form/CustomerForm.php [hunk 1]
```diff
@@ -127,6 +127,13 @@ public function validate()
             }
         }
 
+        $passwordField = $this->getField('password');
+        if (!Validate::isPasswd($passwordField->getValue())) {
+            $passwordField->addError($this->translator->trans(
+                'The password is not in a valid format.', array(), 'Shop.Notifications.Error'
```
[the remark this answers is not in the document]
Comment (Quetzacoalt91) [reply classes/form/CustomerForm.php]: @LouiseBonnard This class is used on the front office. :)

**Comment by Quetzacoalt91** [comment]
I see that the tests do not pass on `should allow filling the personal info form without password`. Is the guest checkout still possible?

File: classes/form/CustomerForm.php [hunk 2]
```diff
@@ -127,6 +127,13 @@ public function validate()
             }
         }
 
+        $passwordField = $this->getField('password');
+        if (!Validate::isPasswd($passwordField->getValue())) {
```
Comment (Quetzacoalt91) [inline classes/form/CustomerForm.php:131]: You should probably check the value is not empty at the same time.

```suggestion
        $passwordValue = $passwordField->getValue();
        if (!empty($passwordValue) && !Validate::isPasswd($passwordValue)) {
```

Comment (PierreRambaud) [reply classes/form/CustomerForm.php:131]: And it maybe should be include in `validateFieldsLengths` instead.

Comment (web-fu) [reply classes/form/CustomerForm.php:131]: Why validateFieldsLenghts? The password validation can be extended for accepting only string with at least "one capital letter, one lowercase letter, one number and one symbol".
The password field validation should be an atomic function. 
I'm not sure that allowing empty password for guest in the same point could a good practice.
Maybe the code should check:
- if the password is mandatory (no guest allowed) OR inserted.
-- TRUE: password have to be valid
-- FALSE: if password is mandatory -> ERROR. Else OK

**Comment by PierreRambaud** [comment]
Closed in favor of #12824

**Pull request closed without merging**